src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -31,15 +31,20 @@ namespace Microsoft.Quantum.QsCompiler.Transformations.Monomorphization
     /// </summary>
     public static class Monomorphize
     {
-        private static bool IsGeneric(QsCallable callable) =>
-            callable.Signature.TypeParameters.Any() || callable.Specializations.Any(spec => spec.Signature.TypeParameters.Any());
-
         /// <summary>
         /// Performs Monomorphization on the given compilation. If the monomorphizeIntrinsics parameter
         /// is set to false, then intrinsics will not be monomorphized.
         /// </summary>
         public static QsCompilation Apply(QsCompilation compilation, bool monomorphizeIntrinsics = false)
         {
+            // If this compilation is for a library project, there are no defined entry points. Instead,
+            // treat every public, non-generic callable as a possible entry point into the library.
+            var hasEntryPoints = compilation.EntryPoints.Length > 0;
+            if (!hasEntryPoints)
+            {
+                compilation = new QsCompilation(compilation.Namespaces, compilation.InteroperableSurface(includeReferences: false).ToImmutableArray());
```

Comment:
What happens here if the library has a non-internal generic? Does it throw? Leave it generic? It seems like that should be a failure case, but perhaps that doesn't trigger a failure until it gets to a generation step that cares about monomorphization having completed.

Reply:
Basically the public generic will be stripped if trimming is enabled (it is by default). What my changes did is precisely making sure that if trimming is disabled for monomorphization, the original generic should just be preserved, and the monomorphized versions will be added with (at most) internal access only (i.e. monormophization no longer silently trims only a part of the trimmable things, which previously resulted in the C# generation bug).